rotkehlchen/db/dbhandler.py

```diff
@@ -2297,6 +2387,9 @@ def query_timed_balances(
                     ),
                 )
 
+        inferred_balances = self._infer_zero_timed_balances(cursor, balances, from_ts, to_ts)
+        balances = list(heapq.merge(balances, inferred_balances, key=lambda x: x.time))
```

Comment:
why use a heapq. Isn't the time complexity `O(NlogN)` in this case with n the number of elements? same as doing `(balances + inferred_balances) .sort()`? also it adds an additional iteration to transform into the list. Just asking if is better or not than a simple sort